File: client/src-tauri/src/commands/contracts/skills.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum SkillSourceKindDto {
    Bundled,
    Local,
    Project,
    Github,
    Dynamic,
    Mcp,
    Plugin,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum SkillSourceScopeDto {
    Global,
    Project,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum SkillSourceStateDto {
    Discoverable,
    Installed,
    Enabled,
    Disabled,
    Stale,
    Failed,
    Blocked,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum SkillTrustStateDto {
    Trusted,
    UserApproved,
    ApprovalRequired,
    Untrusted,
    Blocked,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillSourceMetadataDto {
    pub label: String,
    pub repo: Option<String>,
    pub reference: Option<String>,
    pub path: Option<String>,
    pub root_id: Option<String>,
    pub root_path: Option<String>,
    pub relative_path: Option<String>,
    pub bundle_id: Option<String>,
    pub plugin_id: Option<String>,
    pub server_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum PluginCommandAvailabilityDto {
    Always,
    ProjectOpen,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum PluginCommandRiskLevelDto {
    Observe,
    ProjectRead,
    ProjectWrite,
    RunOwned,
    Network,
    SystemRead,
    OsAutomation,
    SignalExternal,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum PluginCommandApprovalPolicyDto {
    NeverForObserveOnly,
    Required,
    PerInvocation,
    Blocked,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum PluginCommandStatePolicyDto {
    Ephemeral,
    Project,
    Plugin,
    External,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PluginRootDto {
    pub root_id: String,
    pub path: String,
    pub enabled: bool,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PluginDiagnosticDto {
    pub code: String,
    pub message: String,
    pub retryable: bool,
    pub recorded_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PluginSkillContributionDto {
    pub contribution_id: String,
    pub skill_id: String,
    pub path: String,
    pub source_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PluginCommandContributionDto {
    pub command_id: String,
    pub plugin_id: String,
    pub contribution_id: String,
    pub label: String,
    pub description: String,
    pub entry: String,
    pub availability: PluginCommandAvailabilityDto,
    pub risk_level: PluginCommandRiskLevelDto,
    pub approval_policy: PluginCommandApprovalPolicyDto,
    pub state_policy: PluginCommandStatePolicyDto,
    pub redaction_required: bool,
    pub state: SkillSourceStateDto,
    pub trust: SkillTrustStateDto,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PluginRegistryEntryDto {
    pub plugin_id: String,
    pub name: String,
    pub version: String,
    pub description: String,
    pub root_id: String,
    pub root_path: String,
    pub plugin_root_path: String,
    pub manifest_path: String,
    pub manifest_hash: String,
    pub state: SkillSourceStateDto,
    pub trust: SkillTrustStateDto,
    pub enabled: bool,
    pub skill_count: usize,
    pub command_count: usize,
    pub skills: Vec<PluginSkillContributionDto>,
    pub commands: Vec<PluginCommandContributionDto>,
    pub last_reloaded_at: Option<String>,
    pub last_diagnostic: Option<PluginDiagnosticDto>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InstalledSkillDiagnosticDto {
    pub code: String,
    pub message: String,
    pub retryable: bool,
    pub recorded_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillRegistryEntryDto {
    pub source_id: String,
    pub skill_id: String,
    pub name: String,
    pub description: String,
    pub source_kind: SkillSourceKindDto,
    pub scope: SkillSourceScopeDto,
    pub project_id: Option<String>,
    pub source_state: SkillSourceStateDto,
    pub trust_state: SkillTrustStateDto,
    pub enabled: bool,
    pub installed: bool,
    pub user_invocable: Option<bool>,
    pub version_hash: Option<String>,
    pub last_used_at: Option<String>,
    pub last_diagnostic: Option<InstalledSkillDiagnosticDto>,
    pub source: SkillSourceMetadataDto,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillDiscoveryDiagnosticDto {
    pub code: String,
    pub message: String,
    pub relative_path: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillLocalRootDto {
    pub root_id: String,
    pub path: String,
    pub enabled: bool,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillGithubSourceDto {
    pub repo: String,
    pub reference: String,
    pub root: String,
    pub enabled: bool,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillProjectSourceDto {
    pub project_id: String,
    pub enabled: bool,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillSourceSettingsDto {
    pub local_roots: Vec<SkillLocalRootDto>,
    pub plugin_roots: Vec<PluginRootDto>,
    pub github: SkillGithubSourceDto,
    pub projects: Vec<SkillProjectSourceDto>,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SkillRegistryDto {
    pub project_id: Option<String>,
    pub entries: Vec<SkillRegistryEntryDto>,
    pub plugins: Vec<PluginRegistryEntryDto>,
    pub plugin_commands: Vec<PluginCommandContributionDto>,
    pub sources: SkillSourceSettingsDto,
    pub diagnostics: Vec<SkillDiscoveryDiagnosticDto>,
    pub reloaded_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ListSkillRegistryRequestDto {
    pub project_id: Option<String>,
    pub query: Option<String>,
    #[serde(default)]
    pub include_unavailable: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SetSkillEnabledRequestDto {
    pub project_id: String,
    pub source_id: String,
    pub enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RemoveSkillRequestDto {
    pub project_id: String,
    pub source_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct UpsertSkillLocalRootRequestDto {
    pub root_id: Option<String>,
    pub path: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
    pub project_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RemoveSkillLocalRootRequestDto {
    pub root_id: String,
    pub project_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct UpdateProjectSkillSourceRequestDto {
    pub project_id: String,
    pub enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct UpdateGithubSkillSourceRequestDto {
    pub repo: String,
    pub reference: String,
    pub root: String,
    pub enabled: bool,
    pub project_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct UpsertPluginRootRequestDto {
    pub root_id: Option<String>,
    pub path: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
    pub project_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RemovePluginRootRequestDto {
    pub root_id: String,
    pub project_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct SetPluginEnabledRequestDto {
    pub project_id: String,
    pub plugin_id: String,
    pub enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RemovePluginRequestDto {
    pub project_id: String,
    pub plugin_id: String,
}

const fn default_true() -> bool {
    true
}

File: client/src/lib/xero-model/skills.ts
import { z } from 'zod'
import { isoTimestampSchema, nonEmptyOptionalTextSchema, optionalIsoTimestampSchema } from './shared'

export const skillSourceKindSchema = z.enum([
  'bundled',
  'local',
  'project',
  'github',
  'dynamic',
  'mcp',
  'plugin',
])

export const skillSourceScopeSchema = z.enum(['global', 'project'])

export const skillSourceStateSchema = z.enum([
  'discoverable',
  'installed',
  'enabled',
  'disabled',
  'stale',
  'failed',
  'blocked',
])

export const skillTrustStateSchema = z.enum([
  'trusted',
  'user_approved',
  'approval_required',
  'untrusted',
  'blocked',
])

export const pluginCommandAvailabilitySchema = z.enum(['always', 'project_open'])

export const pluginCommandRiskLevelSchema = z.enum([
  'observe',
  'project_read',
  'project_write',
  'run_owned',
  'network',
  'system_read',
  'os_automation',
  'signal_external',
])

export const pluginCommandApprovalPolicySchema = z.enum([
  'never_for_observe_only',
  'required',
  'per_invocation',
  'blocked',
])

export const pluginCommandStatePolicySchema = z.enum(['ephemeral', 'project', 'plugin', 'external'])

export const skillSourceMetadataSchema = z
  .object({
    label: z.string().trim().min(1),
    repo: nonEmptyOptionalTextSchema,
    reference: nonEmptyOptionalTextSchema,
    path: nonEmptyOptionalTextSchema,
    rootId: nonEmptyOptionalTextSchema,
    rootPath: nonEmptyOptionalTextSchema,
    relativePath: nonEmptyOptionalTextSchema,
    bundleId: nonEmptyOptionalTextSchema,
    pluginId: nonEmptyOptionalTextSchema,
    serverId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const installedSkillDiagnosticSchema = z
  .object({
    code: z.string().trim().min(1),
    message: z.string().trim().min(1),
    retryable: z.boolean(),
    recordedAt: isoTimestampSchema,
  })
  .strict()

export const skillRegistryEntrySchema = z
  .object({
    sourceId: z.string().trim().min(1),
    skillId: z.string().trim().min(1),
    name: z.string().trim().min(1),
    description: z.string(),
    sourceKind: skillSourceKindSchema,
    scope: skillSourceScopeSchema,
    projectId: nonEmptyOptionalTextSchema,
    sourceState: skillSourceStateSchema,
    trustState: skillTrustStateSchema,
    enabled: z.boolean(),
    installed: z.boolean(),
    userInvocable: z.boolean().nullable().optional(),
    versionHash: nonEmptyOptionalTextSchema,
    lastUsedAt: optionalIsoTimestampSchema,
    lastDiagnostic: installedSkillDiagnosticSchema.nullable().optional(),
    source: skillSourceMetadataSchema,
  })
  .strict()

export const skillDiscoveryDiagnosticSchema = z
  .object({
    code: z.string().trim().min(1),
    message: z.string().trim().min(1),
    relativePath: nonEmptyOptionalTextSchema,
  })
  .strict()

export const skillLocalRootSchema = z
  .object({
    rootId: z.string().trim().min(1),
    path: z.string().trim().min(1),
    enabled: z.boolean(),
    updatedAt: isoTimestampSchema,
  })
  .strict()

export const pluginRootSchema = z
  .object({
    rootId: z.string().trim().min(1),
    path: z.string().trim().min(1),
    enabled: z.boolean(),
    updatedAt: isoTimestampSchema,
  })
  .strict()

export const pluginDiagnosticSchema = z
  .object({
    code: z.string().trim().min(1),
    message: z.string().trim().min(1),
    retryable: z.boolean(),
    recordedAt: isoTimestampSchema,
  })
  .strict()

export const pluginSkillContributionSchema = z
  .object({
    contributionId: z.string().trim().min(1),
    skillId: z.string().trim().min(1),
    path: z.string().trim().min(1),
    sourceId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const pluginCommandContributionSchema = z
  .object({
    commandId: z.string().trim().min(1),
    pluginId: z.string().trim().min(1),
    contributionId: z.string().trim().min(1),
    label: z.string().trim().min(1),
    description: z.string().trim().min(1),
    entry: z.string().trim().min(1),
    availability: pluginCommandAvailabilitySchema,
    riskLevel: pluginCommandRiskLevelSchema,
    approvalPolicy: pluginCommandApprovalPolicySchema,
    statePolicy: pluginCommandStatePolicySchema,
    redactionRequired: z.boolean(),
    state: skillSourceStateSchema,
    trust: skillTrustStateSchema,
  })
  .strict()

export const pluginRegistryEntrySchema = z
  .object({
    pluginId: z.string().trim().min(1),
    name: z.string().trim().min(1),
    version: z.string().trim().min(1),
    description: z.string().trim().min(1),
    rootId: z.string().trim().min(1),
    rootPath: z.string().trim().min(1),
    pluginRootPath: z.string().trim().min(1),
    manifestPath: z.string().trim().min(1),
    manifestHash: z.string().trim().min(1),
    state: skillSourceStateSchema,
    trust: skillTrustStateSchema,
    enabled: z.boolean(),
    skillCount: z.number().int().nonnegative(),
    commandCount: z.number().int().nonnegative(),
    skills: z.array(pluginSkillContributionSchema).default([]),
    commands: z.array(pluginCommandContributionSchema).default([]),
    lastReloadedAt: optionalIsoTimestampSchema,
    lastDiagnostic: pluginDiagnosticSchema.nullable().optional(),
  })
  .strict()

export const skillGithubSourceSchema = z
  .object({
    repo: z.string().trim().min(1),
    reference: z.string().trim().min(1),
    root: z.string().trim().min(1),
    enabled: z.boolean(),
    updatedAt: isoTimestampSchema,
  })
  .strict()

export const skillProjectSourceSchema = z
  .object({
    projectId: z.string().trim().min(1),
    enabled: z.boolean(),
    updatedAt: isoTimestampSchema,
  })
  .strict()

export const skillRegistryContractDiagnosticSchema = z
  .object({
    code: z.string().trim().min(1),
    message: z.string().trim().min(1),
    severity: z.enum(['info', 'warning', 'error']),
    path: z.array(z.string()),
  })
  .strict()

export const skillSourceSettingsSchema = z
  .object({
    localRoots: z.array(skillLocalRootSchema).default([]),
    pluginRoots: z.array(pluginRootSchema).default([]),
    github: skillGithubSourceSchema,
    projects: z.array(skillProjectSourceSchema).default([]),
    updatedAt: isoTimestampSchema,
  })
  .strict()

export const skillRegistrySchema = z
  .object({
    contractVersion: z.literal(1).default(1),
    projectId: nonEmptyOptionalTextSchema,
    entries: z.array(skillRegistryEntrySchema).default([]),
    plugins: z.array(pluginRegistryEntrySchema).default([]),
    pluginCommands: z.array(pluginCommandContributionSchema).default([]),
    sources: skillSourceSettingsSchema,
    diagnostics: z.array(skillDiscoveryDiagnosticSchema).default([]),
    contractDiagnostics: z.array(skillRegistryContractDiagnosticSchema).default([]),
    reloadedAt: isoTimestampSchema,
  })
  .strict()
  .superRefine((registry, ctx) => {
    const seen = new Set<string>()
    registry.entries.forEach((entry, index) => {
      if (seen.has(entry.sourceId)) {
        ctx.addIssue({
          code: z.ZodIssueCode.custom,
          path: ['entries', index, 'sourceId'],
          message: `Skill registry cannot include duplicate source id \`${entry.sourceId}\`.`,
        })
      }
      seen.add(entry.sourceId)
    })
  })

export const listSkillRegistryRequestSchema = z
  .object({
    projectId: nonEmptyOptionalTextSchema,
    query: nonEmptyOptionalTextSchema,
    includeUnavailable: z.boolean().default(true),
  })
  .strict()

export const setSkillEnabledRequestSchema = z
  .object({
    projectId: z.string().trim().min(1),
    sourceId: z.string().trim().min(1),
    enabled: z.boolean(),
  })
  .strict()

export const removeSkillRequestSchema = z
  .object({
    projectId: z.string().trim().min(1),
    sourceId: z.string().trim().min(1),
  })
  .strict()

export const upsertSkillLocalRootRequestSchema = z
  .object({
    rootId: nonEmptyOptionalTextSchema,
    path: z.string().trim().min(1),
    enabled: z.boolean().default(true),
    projectId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const removeSkillLocalRootRequestSchema = z
  .object({
    rootId: z.string().trim().min(1),
    projectId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const updateProjectSkillSourceRequestSchema = z
  .object({
    projectId: z.string().trim().min(1),
    enabled: z.boolean(),
  })
  .strict()

export const updateGithubSkillSourceRequestSchema = z
  .object({
    repo: z.string().trim().min(1),
    reference: z.string().trim().min(1),
    root: z.string().trim().min(1),
    enabled: z.boolean(),
    projectId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const upsertPluginRootRequestSchema = z
  .object({
    rootId: nonEmptyOptionalTextSchema,
    path: z.string().trim().min(1),
    enabled: z.boolean().default(true),
    projectId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const removePluginRootRequestSchema = z
  .object({
    rootId: z.string().trim().min(1),
    projectId: nonEmptyOptionalTextSchema,
  })
  .strict()

export const setPluginEnabledRequestSchema = z
  .object({
    projectId: z.string().trim().min(1),
    pluginId: z.string().trim().min(1),
    enabled: z.boolean(),
  })
  .strict()

export const removePluginRequestSchema = z
  .object({
    projectId: z.string().trim().min(1),
    pluginId: z.string().trim().min(1),
  })
  .strict()

export type SkillSourceKindDto = z.infer<typeof skillSourceKindSchema>
export type SkillSourceScopeDto = z.infer<typeof skillSourceScopeSchema>
export type SkillSourceStateDto = z.infer<typeof skillSourceStateSchema>
export type SkillTrustStateDto = z.infer<typeof skillTrustStateSchema>
export type SkillSourceMetadataDto = z.infer<typeof skillSourceMetadataSchema>
export type InstalledSkillDiagnosticDto = z.infer<typeof installedSkillDiagnosticSchema>
export type SkillRegistryEntryDto = z.infer<typeof skillRegistryEntrySchema>
export type SkillDiscoveryDiagnosticDto = z.infer<typeof skillDiscoveryDiagnosticSchema>
export type SkillLocalRootDto = z.infer<typeof skillLocalRootSchema>
export type PluginCommandAvailabilityDto = z.infer<typeof pluginCommandAvailabilitySchema>
export type PluginCommandRiskLevelDto = z.infer<typeof pluginCommandRiskLevelSchema>
export type PluginCommandApprovalPolicyDto = z.infer<typeof pluginCommandApprovalPolicySchema>
export type PluginCommandStatePolicyDto = z.infer<typeof pluginCommandStatePolicySchema>
export type PluginRootDto = z.infer<typeof pluginRootSchema>
export type PluginDiagnosticDto = z.infer<typeof pluginDiagnosticSchema>
export type PluginSkillContributionDto = z.infer<typeof pluginSkillContributionSchema>
export type PluginCommandContributionDto = z.infer<typeof pluginCommandContributionSchema>
export type PluginRegistryEntryDto = z.infer<typeof pluginRegistryEntrySchema>
export type SkillGithubSourceDto = z.infer<typeof skillGithubSourceSchema>
export type SkillProjectSourceDto = z.infer<typeof skillProjectSourceSchema>
export type SkillRegistryContractDiagnosticDto = z.infer<typeof skillRegistryContractDiagnosticSchema>
export type SkillSourceSettingsDto = z.infer<typeof skillSourceSettingsSchema>
export type SkillRegistryDto = z.infer<typeof skillRegistrySchema>
export type ListSkillRegistryRequestDto = z.infer<typeof listSkillRegistryRequestSchema>
export type SetSkillEnabledRequestDto = z.infer<typeof setSkillEnabledRequestSchema>
export type RemoveSkillRequestDto = z.infer<typeof removeSkillRequestSchema>
export type UpsertSkillLocalRootRequestDto = z.infer<typeof upsertSkillLocalRootRequestSchema>
export type RemoveSkillLocalRootRequestDto = z.infer<typeof removeSkillLocalRootRequestSchema>
export type UpdateProjectSkillSourceRequestDto = z.infer<typeof updateProjectSkillSourceRequestSchema>
export type UpdateGithubSkillSourceRequestDto = z.infer<typeof updateGithubSkillSourceRequestSchema>
export type UpsertPluginRootRequestDto = z.infer<typeof upsertPluginRootRequestSchema>
export type RemovePluginRootRequestDto = z.infer<typeof removePluginRootRequestSchema>
export type SetPluginEnabledRequestDto = z.infer<typeof setPluginEnabledRequestSchema>
export type RemovePluginRequestDto = z.infer<typeof removePluginRequestSchema>

export function getSkillSourceKindLabel(kind: SkillSourceKindDto): string {
  switch (kind) {
    case 'bundled':
      return 'Bundled'
    case 'local':
      return 'Local'
    case 'project':
      return 'Project'
    case 'github':
      return 'GitHub'
    case 'dynamic':
      return 'Dynamic'
    case 'mcp':
      return 'MCP'
    case 'plugin':
      return 'Plugin'
  }
}

export function getSkillSourceStateLabel(state: SkillSourceStateDto): string {
  switch (state) {
    case 'discoverable':
      return 'Discoverable'
    case 'installed':
      return 'Installed'
    case 'enabled':
      return 'Enabled'
    case 'disabled':
      return 'Disabled'
    case 'stale':
      return 'Stale'
    case 'failed':
      return 'Failed'
    case 'blocked':
      return 'Blocked'
  }
}

export function getSkillTrustStateLabel(state: SkillTrustStateDto): string {
  switch (state) {
    case 'trusted':
      return 'Trusted'
    case 'user_approved':
      return 'User approved'
    case 'approval_required':
      return 'Approval required'
    case 'untrusted':
      return 'Untrusted'
    case 'blocked':
      return 'Blocked'
  }
}

export function getPluginCommandAvailabilityLabel(state: PluginCommandAvailabilityDto): string {
  switch (state) {
    case 'always':
      return 'Always'
    case 'project_open':
      return 'Project open'
  }
}
